src/pages/docs/tutorials/web-midi.md

```diff
@@ -0,0 +1,349 @@
+# Web MIDI
+
+This tutorial explains the basics of controlling an Elementary Audio app with the [Web MIDI API](https://developer.mozilla.org/en-US/docs/Web/API/Web_MIDI_API). The code discussed in this tutorial is available in the MIT-licensed [elementary-webmidi-demo](https://github.com/bgins/elementary-webmidi-demo) repository.
+
+## Playing notes
+We will create a simple web app that plays notes sent by a MIDI controller. The app will:
+
+- Detect when a MIDI device is connected and disconnected
+- Provide an interface for selecting a MIDI device
+- Display the last played MIDI note and its associated frequency
+
+The app uses vanilla JavaScript and no framework to make it broadly accessible. Keep in mind this is a toy application intended to convey ideas. Your app will likely use a framework, stronger types, defensive programming, and other niceties.
+
+At a high level, the app has three parts:
+
+- A Web MIDI module for interfacing with devices
+- Audio engine and synth modules for generating audio graphs and rendering them
+- An event emitter that sends note events from Web MIDI to the audio implementation
+
+The [Web Audio API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API) ultimately renders audio because we are working in the browser.
+
+## Control
+
+Our app uses [WEBMIDI.js](https://webmidijs.org/) to access the Web MIDI API. This library abstracts away many low-level Web MIDI details and makes detecting devices and capturing note events easy.
+
+The [midi module](https://github.com/bgins/elementary-webmidi-demo/blob/main/src/midi.js) includes a `noteEmitter` to send note events and `selectedInput` to track the active MIDI controller. We set the `selectedInput` to `null` until a device is selected.
```

Comment:
Small suggestion: "our" midi module instead of "the" midi module? I think that maybe helps communicate which pieces are part of our demo here versus which pieces are part of "the" web midi API

Reply:
Yeah I like it. Updated this one and the one below. 👌